Userland/Libraries/LibJS/Runtime/IteratorOperations.cpp

```diff
@@ -124,16 +124,22 @@ void iterator_close(Object& iterator)
             vm.unwind(unwind_until, unwind_until_label);
     };
 
-    auto return_method = Value(&iterator).get_method(global_object, vm.names.return_);
-    if (!return_method)
-        return restore_completion(); // If return is undefined, return Completion(completion).
-
-    auto result_or_error = vm.call(*return_method, &iterator);
+    auto return_method_or_error = Value(&iterator).get_method(global_object, vm.names.return_);
+    Value result;
+    if (!return_method_or_error.is_error()) { // If innerResult.[[Type]] is normal, then
```

Comment:
`!is_error()` does not automatically mean [[Type]] is normal - it might implicitly in this case, but let's use `type()` either way.

Reply:
In the current way `ThrowCompletionOr` is implemented, if we have a value (aka if the type is Normal) the `Optional<Completion>` member is empty, so we can actually test the type

Reply:
oh this is ThrowCompletionOr and not Completion, nevermind then